### PR TITLE
Feat/cv orchestrator localize, (GSD Implementation with Unit Tests)

### LINKED ISSUE #20: Integrate Localization
Subtask of #15

TODO writeup

### PR DESCRIPTION
**Pull request by miyatakazuya**
Closes #20 

- Implements GSD-based Localization Algorithm 
- Added Unit Tests from Blender sim test cases
- Currently permits >15ft error, most of the time 1~13ft. 
- Further test cases required for edge cases.

### CODE REVIEW

File: include/cv/localization.hpp
Comment (atar13) [inline include/cv/localization.hpp:87]: nitpick: I wouldn't hate a more descriptive name instead of `CalcOffset` but it's not that big of a deal.

File: src/cv/localization.cpp
Comment (atar13) [inline src/cv/localization.cpp:138]: Fantastic comments on this function. They really make it a lot easier to parse what's going on.

File: src/cv/localization.cpp
Comment (atar13) [inline src/cv/localization.cpp:162]: I really like the inclusion of this function for reasoning about accuracy in terms of feet/meters instead of latitude/longitude.

**Review by atar13 — APPROVED** [review]
Great job on getting this completed and putting in all the work to make sure the code is readable and well documented. Test cases are a great addition that will give us a lot more confidence in our localization algorithm.

This should be ready to merge once you update this branch with the recent changes on `main`. You can do this by running `git fetch` and then `git merge origin/main`. There will be a few conflicting files so feel free to let me know if you're unsure about which changes to keep.

**Comment by miyatakazuya** [comment]
@atar13 What do you want me to do for the interface.hpp ImageTelemetry? I added the heading parameter because the GSD requires it but it is not on the main version of ImageTelemetry.

**Comment by atar13** [comment]
> @atar13 What do you want me to do for the interface.hpp ImageTelemetry? I added the heading parameter because the GSD requires it but it is not on the main version of ImageTelemetry.

We should keep your version with heading. You'll just need to update anywhere the `ImageTelemetry` struct is instantiated to have a heading value. You can verify you fixed everything by running `ninja obcpp` to make sure everything's still compiling.

**Comment by miyatakazuya** [comment]
@atar13 The merge was good and it compiles along with the tests. It didn't pass the format checker so let me know if I should try to resolve that. Otherwise the localization unit test still outputs cout values so if I should remove those lmk but they do help with seeing the current state/accuracy of localization. 

```
...
[----------] 1 test from CVLocalization
[ RUN      ] CVLocalization.LocalizationAccuracy
directly underneath
Error: 0.0162771 feet
Blender Case 1
Error: 13.8251 feet
Blender Case 2
Error: 1.10303 feet
Blender Case 3
Error: 13.2471 feet
Blender Case 4
Error: 12.2687 feet
Blender Case 5
Error: 3.38631 feet
Blender Case 6
Error: 4.55962 feet
[       OK ] CVLocalization.LocalizationAccuracy (0 ms)
[----------] 1 test from CVLocalization (0 ms total)
...
```

**Comment by atar13** [comment]
> @atar13 The merge was good and it compiles along with the tests. It didn't pass the format checker so let me know if I should try to resolve that. Otherwise the localization unit test still outputs cout values so if I should remove those lmk but they do help with seeing the current state/accuracy of localization.
> 
> ```
> ...
> [----------] 1 test from CVLocalization
> [ RUN      ] CVLocalization.LocalizationAccuracy
> directly underneath
> Error: 0.0162771 feet
> Blender Case 1
> Error: 13.8251 feet
> Blender Case 2
> Error: 1.10303 feet
> Blender Case 3
> Error: 13.2471 feet
> Blender Case 4
> Error: 12.2687 feet
> Blender Case 5
> Error: 3.38631 feet
> Blender Case 6
> Error: 4.55962 feet
> [       OK ] CVLocalization.LocalizationAccuracy (0 ms)
> [----------] 1 test from CVLocalization (0 ms total)
> ...
> ```

Yeah, the last thing before merging will be to address the linting/format errors. You can see what it's complaining about [here](https://github.com/tritonuas/obcpp/actions/runs/8855220566/job/24319724290?pr=158) or by running `make lint` in the build directory locally. If you want to ignore any errors you can follow the instructions [here](https://github.com/tritonuas/obcpp?tab=readme-ov-file#best-practices).

I think the prints in the unit test should be fine. Some of the other tests already print to standard out so it's not an issue.

**Pull request merged**
